labelbox/schema/role.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from labelbox.orm.model import Field
+from labelbox.orm.db_object import DbObject
+
+if TYPE_CHECKING:
+    from labelbox.schema.project import Project
+
+
+class Roles:
+    """
+    Object that manages org and user roles
+
+        >>> roles = client.get_roles()
+        >>> roles.valid_roles # lists all valid roles
+        >>> roles['ADMIN'] # returns the admin Role
+
+    """
+    _instance = None
+
+    def __new__(cls, client):
+        if cls._instance is None:
+            cls._instance = super(Roles, cls).__new__(cls)
+            query_str = """query GetAvailableUserRolesPyApi { roles { id name } }"""
+            res = client.execute(query_str)
+            valid_roles = set()
+            for result in res['roles']:
+                _name = result['name'].upper().replace(' ', '_')
+                result['name'] = _name
+                setattr(cls._instance, _name, Role(client, result))
+                valid_roles.add(_name)
+            cls._instance.valid_roles = valid_roles
+        cls._instance.index = 0
+        return cls._instance
+
+    def __repr__(self):
+        return str({k: getattr(self, k) for k in self.valid_roles})
+
+    def __getitem__(self, name):
+        if name in self.valid_roles:
```

Comment:
If you flip this logic it is more concise:


```
if not in names:
    throw
return getattr(self, name)
```

Reply:
fixed